microceph/ceph/osd.go

```diff
@@ -95,22 +95,38 @@ func AddOSD(s *state.State, path string, wipe bool) error {
 	for _, disk := range storage.Disks {
 		// Check if full disk.
 		if disk.Device == dev {
-			path = fmt.Sprintf("/dev/disk/by-id/%s", disk.DeviceID)
+			candidate := fmt.Sprintf("/dev/disk/by-id/%s", disk.DeviceID)
+			// check if candidate exists
+			if shared.PathExists(candidate) {
```

Comment:
Not so sure I understand the code logic. In my original bug report, there wasn't `/dev/disk/by-id` directory, but by mixing up SATA devices and virtio devices there can be a case where `/dev/disk/by-id` does exists but virtio devices requested to wipe doesn't not exist under by-id.

As long as the following command won't be issued, I'm fine with it though.
`dd if=/dev/zero of=/dev/disk/by-id/`